source/server/rorserver.cpp

```diff
@@ -250,7 +255,70 @@ int main(int argc, char* argv[])
 	signal(SIGINT, handler);
 	signal(SIGTERM, handler);
 
-	Sequencer::initilize();
+	// Wait for Listener thread to start before registering on the serverlist
+	//    (which will contact us back for verification).
+
+	// Yay, oldschool pthreads!
+	//    Tutorial: https://computing.llnl.gov/tutorials/pthreads/#ConditionVariables
+
+	pthread_mutex_t listener_ready_mtx;
+	pthread_cond_t  listener_ready_cond;
+	int             listener_ready_value = 0;
+
+	int mtx_result = pthread_mutex_init(&listener_ready_mtx, nullptr);
+	if (mtx_result != 0)
+	{
+		Logger::log(LOG_ERROR, "Failed to initialize mutex (listener_ready_mtx), error code: %d", mtx_result);
+		return -1;
+	}
+
+	int cond_result = pthread_cond_init(&listener_ready_cond, nullptr);
+	if (cond_result != 0)
+	{
+		Logger::log(LOG_ERROR, "Failed to initialize condition-variable (listener_ready_cond), error code: %d", cond_result);
+		return -1;
+	}
+
+	Listener* listener = new Listener(Config::getListenPort(), &listener_ready_mtx, &listener_ready_cond, &listener_ready_value);
+	Sequencer::initialize(listener);
+
+	// Wait for `Listener` to start up.
+	int lock_result = pthread_mutex_lock(&listener_ready_mtx);
+	if (lock_result != 0)
+	{
+		Logger::log(LOG_ERROR, "Failed to acquire lock, error code: %d", lock_result);
+		return -1;
+	}
+	int listener_ready_result = 0;
+	while (listener_ready_value == 0)
+	{
+		listener_ready_result = listener_ready_value;
+		if (listener_ready_result != 0)

```

Comment:
> I tested the code and it worked. Did it fail a test for you?

I did not test it and I don't think that it can fail.

I only question the necessity of that `if block`:
```
		if (listener_ready_result != 0)
		{
			break;
		}
```